src/loaders/muse_load.c

```diff
@@ -0,0 +1,125 @@
+/* Extended Module Player
+ * Copyright (C) 1996-2022 Claudio Matsuoka and Hipolito Carraro Jr
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "loader.h"
+#include "../miniz.h"
+#include "../md5.h"
+
+static void set_md5sum(unsigned char *src, unsigned char *md5, size_t sz)
+{
+	MD5_CTX ctx;
+	MD5Init(&ctx);
+	MD5Update(&ctx, src, sz);
+	MD5Final(md5, &ctx);
+}
```

Comment:
These MD5 hacks in load.c and muse_load.c could probably get lost if there was an `hio_reopen_mem`—what do you think? The depackers could also use this to get rid of the `HIO_HANDLE **`.

Reply:
> if there was an `hio_reopen_mem`

Do you have a draft, or some pseudo code?

Reply:
I'll follow up with a general patch PR + a patch for this branch when I get a chance, sorry. I'm trying to close out #607 but I found a module that potentially complicates it quite a bit, so now I have ~100 of the things to sift through.

Reply:
I implemented this in a branch here: https://github.com/AliceLR/libxmp/commit/b71b779b5f673656a647b4a4d877368ab8d1f689

I'm not sure it ended up being as big of an improvement for the depackers as I hoped, but it should still solve this problem. If it looks acceptable I'll go ahead and open a pull request for it.

Reply:
Looks fair to me at first glance.

Reply:
Rebased and re-adjusted after PR #611